### PR TITLE
update cppcheck + fix new misra violations

### PR DESCRIPTION
**Pull request by adeebshihadeh**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
@briskspirit this PR is getting pretty big. let's break this up into smaller chunks.

File: board/drivers/can.h [hunk 1]
```diff
@@ -55,6 +61,7 @@ can_buffer(tx1_q, 0x100)
 can_buffer(tx2_q, 0x100)
 can_buffer(tx3_q, 0x100)
 can_buffer(txgmlan_q, 0x100)
+// cppcheck-suppress misra-c2012-9.3
```
Comment (robbederks) [inline board/drivers/can.h:64]: Probably fixed by doing `can_ring *can_queues[4] = {&can_tx1_q, &can_tx2_q, &can_tx3_q, &can_txgmlan_q};`

File: board/drivers/can.h [hunk 2]
```diff
@@ -42,10 +44,14 @@ uint32_t ignition_can_cnt = 0U;
...
-int can_live = 0, pending_can_live = 0, can_loopback = 0, can_silent = ALL_CAN_SILENT;
+int can_live = 0;
+int pending_can_live = 0;
+int can_loopback = 0;
+int can_silent = ALL_CAN_SILENT;
 
 // ********************* instantiate queues *********************
-
+// cppcheck-suppress misra-c2012-20.10
+// cppcheck-suppress misra-c2012-20.7
```
Comment (robbederks) [inline board/drivers/can.h:54]: kinda agree with this rule, let's add parentheses

File: board/drivers/can.h [hunk 3]
```diff
@@ -42,10 +44,14 @@ uint32_t ignition_can_cnt = 0U;
...
 
-int can_live = 0, pending_can_live = 0, can_loopback = 0, can_silent = ALL_CAN_SILENT;
+int can_live = 0;
+int pending_can_live = 0;
+int can_loopback = 0;
+int can_silent = ALL_CAN_SILENT;
 
 // ********************* instantiate queues *********************
-
+// cppcheck-suppress misra-c2012-20.10
```
Comment (robbederks) [inline board/drivers/can.h:53]: this one is stupid imo, do we want to suppress it here, or add it to the global suppressions?

File: board/drivers/can.h [hunk 4]
```diff
@@ -65,7 +72,7 @@ int can_err_cnt = 0;
 int can_overflow_cnt = 0;
 
 // ********************* interrupt safe queue *********************
-
+// cppcheck-suppress misra-c2012-8.2
```
Comment (robbederks) [inline board/drivers/can.h:75]: to fix all these, we could declare the prototypes somewhere on top in this file

File: board/drivers/interrupts.h [hunk 5]
```diff
@@ -14,7 +14,7 @@ void unused_interrupt_handler(void) {
 
 #define NUM_INTERRUPTS 102U                // There are 102 external interrupt sources (see stm32f413.h)
 interrupt interrupts[NUM_INTERRUPTS];
-
+// cppcheck-suppress misra-c2012-20.7
```
Comment (robbederks) [inline board/drivers/interrupts.h:17]: let's fix instead of suppress all of the 20.7s

File: board/drivers/can.h (hunk 1 above)
Comment (briskspirit) [reply board/drivers/can.h:64]: Nope, that's what I checked first

File: board/drivers/can.h (hunk 4 above)
Comment (briskspirit) [reply board/drivers/can.h:75]: Prototypes are already declared. It stops complaining if you will remove prototypes, but then you need to shuffle some funcs. Adeeb told to do that as a second PR as it looks massive.

File: board/drivers/interrupts.h (hunk 5 above)
Comment (briskspirit) [reply board/drivers/interrupts.h:17]: They are suppressed till next PR as this one is pretty big already.

File: board/drivers/can.h (hunk 3 above)
Comment (briskspirit) [reply board/drivers/can.h:53]: I would add it to the globals as it makes no sense imho. "The # and ## preprocessor operators should not be used"

Comment (adeebshihadeh) [reply board/drivers/can.h:53]: this one is advisory, but I agree with the rationale and think we should enforce it

File: board/drivers/interrupts.h (hunk 5 above)
Comment (adeebshihadeh) [reply board/drivers/interrupts.h:17]: the diff from 20.7 should be pretty small, let's do it in this PR

File: board/drivers/can.h (hunk 2 above)
Comment (yaddd7) [reply board/drivers/can.h:54]: Adeeb mother tucker

**Pull request closed without merging**